Expose some of the S3 client configuration for assets

KatrinIhler opened the pull request:
This exposes some of the S3 client configuration for assets so we can change it.
This includes:
- Max Connections
- Connection Timeout
- Max Retries

Additionally I improved the related documentation a bit.

KatrinIhler committed:
Make S3 client configurable for assets

Expose S3 client configuration for assets so we change it.
This includes:
- Max Connections
- Connection Timeout
- Max Retries

Additionally improve the related documentation.

gregorydlogan commented:
Can we add these config bits to the distribution code as well?  I can see someone expecting in both places.

KatrinIhler commented:
@gregorydlogan I was afraid you'd ask that...

KatrinIhler committed:
Minor improvements

KatrinIhler committed:
Add S3 client configuration to distribution service

... so we are consistent with the S3 part of the AssetManager.
Also improve the documentation while we're at it.

KatrinIhler commented:
@gregorydlogan Added the same options to distribution and couldn't resist cleaning up the code and the documentation a bit...

gregorydlogan commented on docs/guides/admin/docs/modules/awss3distribution.md:
```diff
@@ -82,18 +82,24 @@ Opencast Service Configuration
+| org.opencastproject.distribution.aws.s3.max.connections              | Number of max connections                                                                | 50                           |                                                                                       |
+| org.opencastproject.distribution.aws.s3.connection.timeout           | Connection timeout in ms                                                                 | 10000                        |                                                                                       |
+| org.opencastproject.distribution.aws.s3.max.retries                  | Number of max retries                                                                    | 100                          |                                                                                       |
+| job.load.aws.s3.distribute                                           | Distribute job load                                                                      | 0.1                          |                                                                                       |
```
So, I hate to be a complainer here, but these aren't exposed on the *other* S3 integration...

Is there a reason you're exposing these?  I don't personally care, but I feel like the chance of someone needing to mess with these values is *so* slim that if they actually *do* need to, then they probably need some kind of larger refactor rather than to mess with the loads.

That being said, if this is something that the sponsor wants I'm cool with it - but it should be in both places :D

KatrinIhler replied:
I didn't expose the job loads, the job loads are already part of the configuration. I just added them to the documentation! Please don't shoot the messenger! :D (I totally agree with you btw, but then, most of the s3 code should probably be shared anyway...)

gregorydlogan reviewed (approved):
Approved pending comment!

mtneug commented on etc/org.opencastproject.distribution.aws.s3.AwsS3DistributionServiceImpl.cfg:
```diff
@@ -47,9 +47,20 @@ org.opencastproject.distribution.aws.s3.distribution.enable=false
 
 # Enable presigned URL
 # Leave this commented out to use presigned URL fow AWS S3
```
I think this is not true. Presigned URLs are not used by default.

```suggestion
```

KatrinIhler replied:
True, I fixed the value, but missed the comment. Thx!

KatrinIhler committed:
Fix comment in S3DistrubutionService config

Co-authored-by: Matthias Neugebauer <mtneug@mailbox.org>
